src/ansys/dpf/core/mesh_info.py

```diff
@@ -0,0 +1,174 @@
+"""
+MeshInfo
+==========
+"""
+import traceback
+
+import ansys.dpf.core
+from ansys.dpf.core import server as server_module
+
+import warnings
+
+
+class MeshInfo:
+    """Represents the mesh information.
+
+    This class describes the available mesh information.
+
+    Parameters
+    ----------
+    mesh_info : ctypes.c_void_p, ansys.grpc.dpf.mesh_info_pb2.MeshInfo message
+
+     server : ansys.dpf.core.server, optional
+        Server with the channel connected to the remote or local instance.
+        The default is ``None``, in which case an attempt is made to use the
+        global server.
+
+    Examples
+    --------
+    Explore the mesh info from the model
+
+    >>> from ansys.dpf import core as dpf
+    >>> from ansys.dpf.core import examples
+    >>> fluent = examples.download_fluent_axial_comp()
+    >>> model = dpf.Model(fluent)
+    >>> mesh_info = model.metadata.mesh_info_provider # printable mesh_info
+
+    """
+
+    def __init__(self, gdc=None, mesh_info=None, server=None):
+        """Initialize with a MeshInfo message"""
+        # ############################
+        # step 1: get server
+        self._server = server_module.get_or_create_server(server)
+
+        if gdc is None:
+            self._gdc = ansys.dpf.core.generic_data_container.GenericDataContainer()
```

Comment:
for the case where we enter with gdc: 

`self._gdc = gdc`

for the mesh_info case:

`self._gdc = mesh_info._gdc`